### ISSUE TITLE
docx writer produces wrong nested enumerated lists from markdown

### ISSUE THREAD

**Issue by m-alles** [jgm/pandoc#6663]
I'm using the latest pandoc 2.10.1 version and this command:

    pandoc test_out.md --from markdown+four_space_rule -o test_docx.docx

Here is the file (please rename .txt to .md).

[test_out.txt](https://github.com/jgm/pandoc/files/5176658/test_out.txt)

The result is that counting is only correct for the top-level list while, the 2nd and 3rd level are never reset when there was a new element of a higher level.

One gets:
    1.
     (a)
      i.
      ii.
      iii.
     (b)
      iv.
      v.
     2.
     (c)
 
Obviously, below the (b) one should start counting from i. again and below the 2. one should start counting with (a) again.

A quick look into the docx file shows that for i. to v. the same abstractNumId is being used for the list elements below (a) and (b). When introducing a new one for the roman list below (b) one can fix this issue.

The .odt export works nicely in that regard by the way.

Thanks.

**Comment by jgm** [comment]
Can't reproduce this.

Here's what I get with your sample (viewing it in Word -- are you using something else?)

<img width="799" alt="Screen Shot 2020-09-05 at 11 27 50 AM" src="https://user-images.githubusercontent.com/3044/92311320-dc4fa380-ef6a-11ea-8528-95d29e87d296.png">

The main problem here is that all the lists are on the same indentation level.  But that is because the indentation isn't right in your input markdown file.  (See the manual on the required indentation for sublists.)  When that is fixed

[fixed.txt](https://github.com/jgm/pandoc/files/5178485/fixed.txt)

we get this:

<img width="781" alt="Screen Shot 2020-09-05 at 11 29 58 AM" src="https://user-images.githubusercontent.com/3044/92311361-28024d00-ef6b-11ea-9002-9192b11fd423.png">

which looks just right to me.  I conclude that there is no bug in pandoc.  Perhaps if you're looking at the docx using LibreOffice or another program, there is a bug in its interpretation of docx?

**Comment by m-alles** [comment]
Hi,

Thanks for looking into this. The indentation was not an issue for me as I use "--from markdown+four_space_rule" for converting.

Indeed, I didn't check it in Word itself but used Libreoffice 6.4. With Libreoffice 7.0 as well as Google Docs and Words, the document looks good. However, all three show the same behavior when I try to make changes.

For instance: If I add 1.(b).iii that works (by pressing the Return butten at the end of 1.(b).ii. If I then use Shift+<tab> to decrease the indentation level it gives ii. instead of (c).
Same If I add 2.(b) that works. If I then use <tab> I get (a) instead of i. If I add tab another time, it still says (a).

So while displaying is correct the structure in the document itself seems not consistent. I'm not sure about the goal of pandoc in that regard, but I assume that editing a file after conversion should be feasible.

Thanks,
Matthias

**Comment by m-alles** [comment]
Should say "shift+tab" not "shift+"

**Comment by jgm** [comment]
Thanks for the clarification. That does seem to be a potential problem, which I'll need to look at further.
(I haven't yet tried to reproduce.)

**Comment by jgm** [comment]
Yes, I can reproduce this.  When you press shift-tab, it reduces the indentation level of the list while keeping the same number.  We might be able to do better in this regard.

**Comment by m-alles** [comment]
I analysed this issue somewhat further.

Here is the structure that pandoc creates:

NumId => abstractNumId => Section
1001 => 99411 => 1., 2.
1002 => 99731 => 1.(a), 1.(b)
1003 => 99511 => 1.(a).i, 1.(a).ii, 1.(a).iii
1004 => 99511 (99512) => 1.(b).i, 1.(b).ii
1005 => 99412
1006 => 99731 (99732) => 2.(a)

Here is the structure that Word creates:
1 => 0 => any section

The different abstractNumIds of pandoc contain for each indentation level the same style, e.g. 99411 has the style of decimal numbers specified for all indentation levels. The abstractNumId 99731 has lowerLetter defined for all indentation levels, and so on.

In Word, the abstractNumId has decimal number defined for level 0, lowerLetter for level 1, lowerRoman for level 2, and so on.

So it seems the issue is the "more than necessary" complexity of the docx file generated by pandoc.

**Comment by jgm** [comment]
One reason we can't do it that way is that in pandoc, a list could have any arbitrary styling at any level.

**Comment by m-alles** [comment]
Would it be possible to set the style of the different indentation levels in the abstractNumIds differently, i.e. abstractNumId 99411 would have style decimal for level 0, style lowerLetter for indentation level 1, lowerRoman for level 2, etc? In this way, the rendered document would look the same and becomes editable.

**Comment by jgm** [comment]
How would we determine what the other level styles should be, though?  I guess we'd have to analyze the entire containing list.

**Comment by agusmba** [comment]
1. one
   a) two
   b) three
2. four
   i) five
   ii) six

And since Markdown lets you mix things up, it can be easy to create lists that are not supported in Word in one single list style (different formats at the same depth)

**Comment by m-alles** [comment]
I wonder how this is handled in the OpenDocument Text writer. There, the output works better in this regard, but it's not perfect either.

**Comment by m-alles** [comment]
> And since Markdown lets you mix things up, it can be easy to create lists that are not supported in Word in one single list style (different formats at the same depth)

But at least you know the styles of the higher indentation levels.
